cpp/include/algorithms.hpp

```diff
@@ -622,9 +622,8 @@ void bfs(raft::handle_t const &handle,
  *                                   Supported value : int (signed, 32-bit)
  * @tparam weight_t                  Type of edge weights. Supported values : float or double.
  *
+ * @param[in]  handle                Library handle (RAFT). If a communicator is set in the handle,
  * @param[in]  graph                 input graph object (CSR)
- * @param[out] final_modularity      modularity of the returned clustering
- * @param[out] num_level             number of levels of the returned clustering
  * @param[out] clustering            Pointer to device array where the clustering should be stored
```

Comment:
Agree the names should be consistent.  I'm slightly more inclined to change louvain_parts to clustering, but could be talked into either one (or something different).

Regarding the allocation question, this is analogous to the pagerank result.  The current paradigm we use in the interface is to allocate memory when we know the structure of the return a priori in the python/cython layer and pass down the pointer.  Currently we only allocate memory in the C++ and return it if the exact size/shape of the memory required is unknown until during the execution of the algorithm.

Are you suggesting that we change the overall paradigm?  We could certainly have all of the algorithms (Louvain, Pagerank, etc) allocate their result array in C++ and return it.

I moved the final_modularity and num_level (which are a single value of type weight_t and int, respectively) into the pair to return to eliminate the pass by reference/address discrepancy between Louvain and Leiden.  But if we think it's better for all of the algorithms to allocate their return device vectors and return them I can blaze the trail in Louvain.

Reply:
I'm fine with either clustering or louvain_parts but comments and the code should agree.

Yeah... I don't suggest changing the overall paradigm. Returning 2+ variables using std::pair (or std::tuple) while still returning some results through input pointers looks a bit unorthodox, but this may be the best at this point.

May discuss how to resolve this later.

Reply:
Done